### PR TITLE
Updates Documentation for Elasticsearch

### PR DESCRIPTION
**Pull request by JeffNaef-Triggermesh**
No FE updates needed :)

### CODE REVIEW

File: docs/targets/elasticsearch.md [hunk 1]
```diff
@@ -1,75 +1,58 @@
-# Event target for Elasticsearch
+# Event Target for Elasticsearch
 
-This event target receives [CloudEvents][ce] over HTTP and index their payload intoElasticsearch.
+This event Target receives [CloudEvents][ce] over HTTP and indexs their payload into [Elasticsearch][es]
```
Comment (antoineco) [inline docs/targets/elasticsearch.md:3]: Typo

File: docs/targets/elasticsearch.md [hunk 2]
```diff
@@ -1,75 +1,58 @@
...
 
 ## Prerequisites
 
-A Elasticsearch cluster and a set of credentials for indexing documents.
+1. If there is not a pre-existing Elasticsearch cluster available, get started quickly by using either [Elastic Cloud on Kubernetes][eck] or [Elastic Cloud][elasticcloud]
```
Comment (antoineco) [inline docs/targets/elasticsearch.md:7]: "There is no"

File: docs/targets/elasticsearch.md [hunk 3]
```diff
@@ -1,75 +1,58 @@
...
 
-You can get started by using:
+1. Retrieve the proper credentials in either of the following forms:
+    * User and password to the Elasticsearch cluster.
+    * An APIKey instead of User and password.
```
Comment (antoineco) [inline docs/targets/elasticsearch.md:11]: "API key"

File: docs/targets/elasticsearch.md (hunk 1 above)
Comment (antoineco) [reply docs/targets/elasticsearch.md:3]: Missing dot.

File: docs/targets/elasticsearch.md (hunk 2 above)
Comment (antoineco) [reply docs/targets/elasticsearch.md:7]: Missing dot.

File: docs/targets/elasticsearch.md (hunk 3 above)
Comment (antoineco) [reply docs/targets/elasticsearch.md:11]: Lowercase user.

File: docs/targets/elasticsearch.md [hunk 4]
```diff
@@ -1,75 +1,58 @@
...
 
-- [Elastic Cloud on Kubernetes][eck]
-- [Elascic Cloud][elasticcloud]
+Consort the [Elasticsearch documentation][docs] for more information on Elasticsearch.
```
Comment (antoineco) [inline docs/targets/elasticsearch.md:13]: "Consort"?

File: docs/targets/elasticsearch.md [hunk 5]
```diff
@@ -1,75 +1,58 @@
...
 
-## Create the Elasticsearch target integration
+## Deploying an Instance of the Target
 
-Create the Elasticsearch target integration in 2 steps:
+Open the Bridge creation screen and add a Target of type `Elastic`.
```
Comment (antoineco) [inline docs/targets/elasticsearch.md:17]: "Elasticsearch" (Elastic is the company that builds Elasticsearch).

File: docs/targets/elasticsearch.md [hunk 6]
```diff
@@ -1,75 +1,58 @@
...
 
-1. Retrieve credentials for Elasticsearch.
-2. Create the Elasticsearch target.
+![Adding a Elasticsearch Target](../images/elastic-target/create-bridge-1.png)
```
Comment (antoineco) [inline docs/targets/elasticsearch.md:19]: Please rename the directory to "elasticsearch-target" (Elastic is the company that builds Elasticsearch).

File: docs/targets/elasticsearch.md (hunk 2 above)
Comment (antoineco) [reply docs/targets/elasticsearch.md:7]: @cab105 do we need to replace the 2nd person in this imperative? If we do, the whole page needs a rewrite.

See also my comment on #34.

File: docs/targets/elasticsearch.md [hunk 7]
```diff
@@ -1,75 +1,58 @@
...
 
-### Elasticsearch credentials
+In the Target creation form, give a name to the event Target and add the following information:
 
-Credentials for Elasticsearch  which could be in the form:
+* **Index** : Location the documents will be indexed.
```
Comment (antoineco) [inline docs/targets/elasticsearch.md:23]: "Nane of the index to write documents to".

File: docs/targets/elasticsearch.md [hunk 8]
```diff
@@ -1,75 +1,58 @@
...
+* **Addresses** : Server address list.
```
Comment (antoineco) [inline docs/targets/elasticsearch.md:24]: "List of URLs of Elasticsearch servers."

File: docs/targets/elasticsearch.md (hunk 1 above)
Comment (sebgoa) [inline docs/targets/elasticsearch.md:3]: indexes

Comment (sebgoa) [reply docs/targets/elasticsearch.md:3]: period

File: docs/targets/elasticsearch.md (hunk 4 above)
Comment (sebgoa) [inline docs/targets/elasticsearch.md:13]: Check the docs ...

File: docs/targets/elasticsearch.md [hunk 9]
```diff
@@ -1,75 +1,58 @@
...
+* **Skip verify** : Allows skipping the server certificate verification.
+* **CA certificate** : For adding the PEM string for the certificate.
```
Comment (antoineco) [inline docs/targets/elasticsearch.md:26]: Just "CA certificate to be used by the event target's client, in PEM format."

File: docs/targets/elasticsearch.md [hunk 10]
```diff
@@ -1,75 +1,58 @@
...
+When using a self-signed certificate you will need to either inform the **CA certificate** or set the **Skip verify** field.
 
-Additionaly when using a self signed certificate you will need to either inform the `CACertificate` or set the `SkipVerify` field.
+![Elastic target form](../images/elastic-target/create-bridge-2.png)
 
-### Deploy Elasticsearch target
+After clicking the `Save` button, the console will self-navigate to the Bridge editor. Proceed by adding the remaining components to the Bridge.
 
-At Triggermesh add a new secret depending on the credentials set:
+After submitting the bridge, and allowing some configuration time,  green check mark on the main _Bridges_ page indicates that the bridge with an Elasticsearch event Target was successfully created.
```
Comment (sebgoa) [inline docs/targets/elasticsearch.md:37]: bridge or Bridge ?

Comment (sebgoa) [reply docs/targets/elasticsearch.md:37]: extra space after "time, "

File: docs/targets/elasticsearch.md [hunk 11]
```diff
@@ -1,75 +1,58 @@
...
-### Elasticsearch credentials
+In the Target creation form, give a name to the event Target and add the following information:
 
-Credentials for Elasticsearch  which could be in the form:
+* **Index** : Location the documents will be indexed.
+* **Addresses** : Server address list.
+* **Skip verify** : Allows skipping the server certificate verification.
+* **CA certificate** : For adding the PEM string for the certificate.
+* **Username**: Elasticsearch username.
+* **Password**:  Reference a [TriggerMesh secret][tm-secret] containing a Password to communicate with the Elasticsearch API, as described in the previous section.
```
Comment (antoineco) [inline docs/targets/elasticsearch.md:28]: Lowercase password.

File: docs/targets/elasticsearch.md (hunk 10 above)
Comment (antoineco) [reply docs/targets/elasticsearch.md:37]: **a** green check mark.

File: docs/targets/elasticsearch.md [hunk 12]
```diff
@@ -1,75 +1,58 @@
...
-curl -v http://elasticsearchtarget \
- -X POST \
- -H "Content-Type: application/json" \
- -H "Ce-Specversion: 1.0" \
- -H "Ce-Type: my.data.type" \
- -H "Ce-Source: some.origin/intance" \
- -H "Ce-Id: 536808d3-88be-4077-9d7a-a3f162705f79" \
- -d '{"message":"thanks for indexing this message","from": "Triggermesh targets", "some_number": 12}'
-```
+This Target expects a [JSON][ce-jsonformat] payload with the following properties:
```
Comment (antoineco) [inline docs/targets/elasticsearch.md:47]: That is in direct contradiction with the above sentence.

File: docs/targets/elasticsearch.md [hunk 13]
```diff
@@ -1,75 +1,58 @@
...
+| Name  |  Type |  Comment |
+|---|---|---|
+| **message**| string  |  |
+|  **from** |  string |  |
```
Comment (antoineco) [inline docs/targets/elasticsearch.md:51]: Extra leading space.

File: docs/targets/elasticsearch.md (hunk 1 above)
Comment (cab105) [reply docs/targets/elasticsearch.md:3]: indices instead of indexes.

File: docs/targets/elasticsearch.md (hunk 2 above)
Comment (cab105) [reply docs/targets/elasticsearch.md:7]: More than likely not in this case, however as per your comment https://github.com/triggermesh/docs/pull/29/files#r466502571 this should be consistent in the usage.

File: docs/targets/elasticsearch.md (hunk 4 above)
Comment (cab105) [reply docs/targets/elasticsearch.md:13]: I'm sure he meant consult.

File: docs/targets/elasticsearch.md [hunk 14]
```diff
@@ -1,75 +1,56 @@
-# Event target for Elasticsearch
+# Event Target for Elasticsearch
 
-This event target receives [CloudEvents][ce] over HTTP and index their payload intoElasticsearch.
+This event Target receives [CloudEvents][ce] over HTTP and indices their payload into [Elasticsearch.][es]
```
Comment (antoineco) [inline docs/targets/elasticsearch.md:3]:
```suggestion
This event Target receives [CloudEvents][ce] over HTTP and writes their payload to an [Elasticsearch][es] index.
```

File: docs/targets/elasticsearch.md [hunk 15]
```diff
@@ -1,75 +1,56 @@
...
 
 ## Prerequisites
 
-A Elasticsearch cluster and a set of credentials for indexing documents.
+1. If there is no pre-existing Elasticsearch cluster available, get started quickly by using either [Elastic Cloud on Kubernetes][eck] or [Elastic Cloud.][elasticcloud]
```
Comment (antoineco) [inline docs/targets/elasticsearch.md:7]:
```suggestion
1. If there is no pre-existing Elasticsearch cluster available, get started quickly by using either [Elastic Cloud on Kubernetes][eck] or [Elastic Cloud][elasticcloud].
```

File: docs/targets/elasticsearch.md [hunk 16]
```diff
@@ -1,75 +1,56 @@
...
 
-You can get started by using:
+1. Retrieve the proper credentials in either of the following forms:
+    * User and password to the Elasticsearch cluster.
+    * An API Key instead of user and password.
```
Comment (antoineco) [inline docs/targets/elasticsearch.md:11]:
```suggestion
1. Retrieve credentials for the Elasticsearch cluster in either of the following forms:
    * User and password
    * API Key
```

File: docs/targets/elasticsearch.md [hunk 17]
```diff
@@ -1,75 +1,56 @@
...
-2. Create the Elasticsearch target.
+![Adding a Elasticsearch Target](../images/elasticsearch-target/create-bridge-1.png)
 
-### Elasticsearch credentials
+In the Target creation form, give a name to the event Target and add the following information:
 
-Credentials for Elasticsearch  which could be in the form:
+* **Index** : Name of the index to write documents to.
+* **Addresses** : List of URLs of Elasticsearch servers.
+* **Skip verify** : Allows skipping the server certificate verification.
```
Comment (antoineco) [inline docs/targets/elasticsearch.md:25]:
```suggestion
* **Skip verify** : Allow skipping the server certificate verification.
```

File: docs/targets/elasticsearch.md [hunk 18]
```diff
@@ -1,75 +1,56 @@
...
- -H "Ce-Source: some.origin/intance" \
- -H "Ce-Id: 536808d3-88be-4077-9d7a-a3f162705f79" \
- -d '{"message":"thanks for indexing this message","from": "Triggermesh targets", "some_number": 12}'
-```
+The Elasticsearch event Target expects a [JSON][ce-jsonformat] payload with the following properties:
+| Name  |  Type |  Comment |
+|---|---|---|
+|**message**|string||
+|**from**|string||
+|**some_number**|int||
```
Comment (antoineco) [inline docs/targets/elasticsearch.md:50]:
```suggestion
The Elasticsearch event target can consume events of any type.
```

**Comment by antoineco** [comment]
I did a few edits to speed this up, I believe the doc is in a good shape now. @sebgoa care to stamp it?

**Comment by sebgoa** [comment]
merging this, thanks

**Pull request merged**